activerecord/test/cases/serialized_attribute_test.rb

```diff
@@ -240,6 +240,20 @@ def test_regression_serialized_default_on_text_column_with_null_false
     assert_equal [], light.long_state
   end
 
+  def test_unexpected_serialized_type
+    Topic.serialize :content, Hash
+    topic = Topic.create!(content: { zomg: true })
+
+    Topic.serialize :content, Array
```

Comment:
It won't because https://github.com/kirs/rails-1/blob/serialized-error-attribute/activerecord/test/cases/serialized_attribute_test.rb#L14-L16